pkg/oauth/apis/oauth/types.go

```diff
@@ -4,6 +4,9 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// +genclient
+// +genclient:nonNamespaced
```

Comment:
Not sure I understand what's broken about list... also, what's not covered by the generic registry implementation? Patch is the only thing I haven't found so far